pip/__init__.py

```diff
@@ -209,6 +210,9 @@ def main(args=None):
         sys.stderr.write(os.linesep)
         sys.exit(1)
 
+    # Needed for locale.getpreferredencoding(False) to work
+    # in pip.utils.encoding.auto_decode
+    locale.setlocale(locale.LC_ALL, '')
```

Comment:
Some tools, which use pip's python API, fails too. May be it is better to call `getpreferredencoding(True)`?